[T09-1] MakerManager

trufflepirate opened the pull request:
Ready for Review

@Psyf
@teojunjie
@Ftywan

nishantbudhdev commented:
@trufflepirate @teojunjie @Psyf @Ftywan 

Overall good progress in updating the documents.

## User guide:

- Remove reference to addressbook in Section 2.
- Remove reference to addressbook in the download link.

## Developer guide:

- Some updated diagrams still refer to Addressbook. Consider renaming them to reflect MakerManager.

## Code:

- Is `public` the best access modifier for all data members? As a rule of thumb start all data members as private, and depending on their usage promote them to protected, and if absolutely necessary, to public.

Psyf committed:
Change symbol constraints in addAdmin

teojunjie committed:
Updated ppp , developer and user guide

Psyf committed:
manageJobCommandParser input sanitization against invalid JobName

Psyf committed:
delete Job re-enabled

Psyf committed:
Auto addition fix

Ftywan committed:
testing

Ftywan committed:
Merge branch 'master' into testing

Ftywan committed:
Merge remote-tracking branch 'origin/testing' into tytest

Ftywan committed:
Merge pull request #23 from CS2113-AY1819S1-T09-1/master

keep updated with the master branch

Ftywan committed:
fix travis

teojunjie committed:
ReWrote logic to check if machine is allowed to be editable

teojunjie committed:
Resolving travis errors

teojunjie committed:
Resolving travis errors

teojunjie committed:
Added remove machine command without proper implementation only added the backbone for parser and logic

teojunjie committed:
Wrote tests for machinelist and equality to increase coverage

teojunjie committed:
Resolving travis errors

teojunjie committed:
Resolving travis erros

teojunjie committed:
Merge branch 'jj1.4issues' into jj1.4features

teojunjie committed:
Trying to improve coverage

teojunjie committed:
Trying to improve coverage

Ftywan committed:
travis

Ftywan committed:
Merge remote-tracking branch 'origin/master'

Ftywan committed:
testing cases

teojunjie committed:
Wrote tests in unique machine list

teojunjie committed:
Resolving travis errors

Ftywan committed:
Merge pull request #84 from Ftywan/testing

added testing codes for the job model

teojunjie committed:
Removed failing tests

Ftywan committed:
Merge remote-tracking branch 'org/master'

teojunjie committed:
Merge branch 'master' into jj1.4issues

Ftywan committed:
Revert "added testing codes for the job model"

teojunjie committed:
Merge branch 'jj1.4issues' into jj1.4features

Ftywan committed:
Merge pull request #135 from CS2113-AY1819S1-T09-1/revert-84-testing

Revert "added testing codes for the job model"

Ftywan committed:
Merge remote-tracking branch 'org/master'

teojunjie committed:
Merge branch 'master' into jj1.4issues

Ftywan committed:
fix bugs from pe1

Ftywan committed:
Merge pull request #136 from Ftywan/pebug

bug for pe-1

Psyf committed:
Review fix: remove TODO

Commit:
Merge branch 'master' into psyf1.4.1Fixes

Commit:
Merge branch 'master' into jj1.4issues

Commit:
Merge pull request #127 from teojunjie/jj1.4issues

Resolving v1.4 issues from PE1 by JJ

Psyf committed:
Admin UI more apparent :: commandBox style changed

trufflepirate committed:
SwapJobs fixed

trufflepirate committed:
Merge pull request #129 from Psyf/psyf1.4.1Fixes

Psyf's fixes for v1.4.1

trufflepirate committed:
Merge remote-tracking branch 'UPSTREAM/master' into dev1_4_2

teojunjie committed:
Merge branch 'master' into jj1.4features

teojunjie committed:
Added manageMachine parser and command, not integrated yet

teojunjie committed:
Added flush and remove commands, havent tested it out yet

teojunjie committed:
Added clean function for machine , integrated but not tested yet

teojunjie committed:
Travis checkstyle

teojunjie committed:
Added prompt for user to determine if he wants to execute the flush

teojunjie committed:
Added auto flush option but have not integrated with saif auto code yet

teojunjie committed:
Resolving travis errors

teojunjie committed:
Resolving travis errors

teojunjie committed:
Resolving travis errors

teojunjie committed:
Added tests to improve coverage

teojunjie committed:
Resolving travis errors

trufflepirate committed:
Swap and move command.
+ modified addressbook instance saving to deepcopy all/most child elements.

teojunjie committed:
Removed tests

Psyf committed:
manageJob_start_restart only applies for top of queue

teojunjie committed:
Working auto move machines

teojunjie committed:
Resolving travis errors

teojunjie committed:
Resolving travis errors

Psyf committed:
Fixing tests, commenting out 1.

Commit:
Merge pull request #134 from teojunjie/jj1.4features

Jj1.4features

Commit:
Merge branch 'master' into psyf1.4.2Dev

Psyf committed:
Fix merge error

teojunjie committed:
Added fuzzy wuzzy library and playing around with it

trufflepirate committed:
Fixed travis errors

Psyf committed:
Fix Travis StyleChecks

Psyf committed:
So stringent these checks

trufflepirate committed:
Fixed AddjobtoMachineTest breaking
Added constraints to ongoing Job

trufflepirate committed:
Added Ongoing job constraint to SwapJobs

teojunjie committed:
Working fuzzy wuzzy implementation

teojunjie committed:
Resolving travis errors

teojunjie committed:
Resolving travis errors

teojunjie committed:
Update UniqueMachineList.java

Remove unwanted javadoc statement

teojunjie committed:
Merge branch 'master' into jj1.4featuresv2

teojunjie committed:
Merge branch 'jj1.4featuresv2' of https://github.com/teojunjie/main into jj1.4featuresv2

trufflepirate committed:
Merge branch 'master' into dev1_4_2

teojunjie committed:
Updated userguide

teojunjie committed:
Corrected error in ugdg

teojunjie committed:
Changed flow of order of commands according to friend's suggestion

trufflepirate committed:
Merge Fixes

trufflepirate committed:
Travis

Psyf committed:
Standardize UG, add Upcoming and Parting Words Sections.

Psyf committed:
Redone UG according to Checklist specifications, still WIP

Psyf committed:
UG DG polish according to checklist::2101

Psyf committed:
Added Problem section under Features for some background

Psyf committed:
Styling

teojunjie committed:
Added ability to auto add to different machines instead of just one

teojunjie committed:
Resolving travis errors

Psyf committed:
Added Notes to UG; started work on DG

teojunjie committed:
Merge pull request #146 from Psyf/psyf1.4.2Dev

Psyf1.4.2 dev

Psyf committed:
Change DG according to 2101 slides

Ftywan committed:
adding testing files

Ftywan committed:
adding testing files

Ftywan committed:
[WIP] resolving the testing cases.

teojunjie committed:
Remove fuzzy wuzzy code for findMachine

teojunjie committed:
Update build.gradle

Took away fuzzy wuzzy dependency

teojunjie committed:
Remove unnecessary ui elements

teojunjie committed:
Updated dg diagrams plus dg

teojunjie committed:
Rename method according to saif suggestion

teojunjie committed:
Resolving travis errors

Commit:
Merge pull request #152 from teojunjie/jj1.4featuresv2

More like a miscomm bug fix.

trufflepirate committed:
Multiple Bugfixes and Integration upgrades:
+Ui changes
+Swap/Move commands
+fixed refreshing bug
+fixed undo bug

trufflepirate committed:
Fixed Bug in reSorting Logic

trufflepirate committed:
Merge branch 'master' into dev1_4_2

trufflepirate committed:
Reimplemented top job constraint

Commit:
Merge pull request #151 from trufflepirate/dev1_4_2

Dev1 4 2

Ftywan committed:
adding more testing cases.

trufflepirate committed:
Multiple Changes:
+added Constraint: cannot add Job to disabled Machines
+fixed bug with isTopJob failing when empty machine
+fixed regression bug with undo/redo not saving changes in Job

Commit:
Merge pull request #157 from trufflepirate/dev1_4_3

Multiple Changes:

trufflepirate committed:
Updated UG with added commands

trufflepirate committed:
Fixed Minor Format Error

teojunjie committed:
Did parts from 2.1 to 2.5 and added some new pictures

teojunjie committed:
Changed pic of addmachineparserlogicmodel.png

Ftywan committed:
add testing for AddJobCommand

Ftywan committed:
adding testing cases

Ftywan committed:
pass building tests

Ftywan committed:
cleans up the format of UG, reorders the format

Ftywan committed:
fix travis tests

Ftywan committed:
fix travis errors

Ftywan committed:
fix travis errors

Ftywan committed:
fix travis

Ftywan committed:
fix travis errors

Ftywan committed:
checked and updated the UGDG

trufflepirate committed:
Minor Ui Upgrades

trufflepirate committed:
Fixed a bug where the lists will not re-shuffle when new job is added

teojunjie committed:
Added my own feature : storage system for maker manager (jun jie feature) to be used in my own ppp

teojunjie committed:
Merged v1.4ugdg into jjportfoliobranch

teojunjie committed:
Added tags for user guide and developer guide to be use in my (jun jie ppp)

teojunjie committed:
Removed accidentally commited html files'

teojunjie committed:
Changed picture of storage component, had a big blank space for the old one

teojunjie committed:
Added stuff for my own ppp

Commit:
Delete data\addressbook.xml

Commit:
Delete data\makerManagerAdmins.xml

Commit:
Delete data\makerManagerMachines.xml

Commit:
Merge pull request #158 from Ftywan/testing

Testing codes for job relevant classes

Psyf committed:
Timer to funct

Psyf committed:
Remove some code

Psyf committed:
Removed some TODO

Commit:
Update config.json

Commit:
Update preferences.json

Psyf committed:
Travis fix

Psyf committed:
Merge branch 'refactor-1.4' of https://github.com/Psyf/main into refactor-1.4

Psyf committed:
Merge branch 'v1.4ugdg' of https://github.com/teojunjie/main into v1.4ugdg

Psyf committed:
Add new tags for PPP

Psyf committed:
small changes

trufflepirate committed:
Changed addMachine Test, added default model stub

trufflepirate committed:
Merge remote-tracking branch 'UPSTREAM/master' into dev1_4_4

# Conflicts:
#	preferences.json

Psyf committed:
Work on PPP, DG, UG and refactor Log/in/out

teojunjie committed:
Added manual testing section part abit

teojunjie committed:
Merge branch 'v1.4ugdg' of https://github.com/teojunjie/main into v1.4ugdg

Commit:
Merge branch 'master' into v1.4ugdg

teojunjie committed:
Merge branch 'v1.4ugdg' into jjportfoliobranch

teojunjie committed:
Merge branch 'v1.4ugdg' of https://github.com/teojunjie/main into v1.4ugdg

Commit:
Update config.json

Commit:
Update UserGuide.adoc

Commit:
Update DeveloperGuide.adoc

Commit:
Update preferences.json

teojunjie committed:
Added sample data files by removing '/data/' line in gitignore

teojunjie committed:
Updated dg with appropriate instructions to start application along with data sets

teojunjie committed:
Removed old DemoDatav1.4.zip by saif

teojunjie committed:
Added back DemoDatav1.4.zip

teojunjie committed:
Merge branch 'v1.4ugdg' into jjportfoliobranch

teojunjie committed:
Added instructions in dg to start application by downloading demodata.zip file and unzipping it into data folder

Psyf committed:
PPP

teojunjie committed:
Added instructions for quickstart of ug, and changed the Ui.png image to a more updated version with the revelant sample data

teojunjie committed:
Resolved conflicts in jjportfoliobranch

Psyf committed:
Merge branch 'v1.4ugdg' of https://github.com/teojunjie/main into v1.4ugdg

teojunjie committed:
Updated teojunjie.adoc ppp

teojunjie committed:
Merge branch 'v1.4ugdg' of https://github.com/teojunjie/main into v1.4ugdg

teojunjie committed:
Update developer guide 'loading sample data'

teojunjie committed:
Delete addressbook.xml

teojunjie committed:
Delete makerManagerAdmins.xml

teojunjie committed:
Delete makerManagerMachines.xml

teojunjie committed:
Added '/data/' option back to gitignore

teojunjie committed:
Updated teojunjie.adoc

trufflepirate committed:
Added Tests and Squashed bugs:
+AddMachine Command Test rework
+AddMachine Command Parser Test
+editMachine Command Test
+editMachine Command Parser Test

+Fixed Bug related editMachine allowing name Change to AUTO

trufflepirate committed:
Merge pull request #160 from Psyf/refactor-1.4

Some small adjustments

trufflepirate committed:
Changed error message in EditMachineCommand

trufflepirate committed:
Changed Success message of editMachine

trufflepirate committed:
fixing Travis

trufflepirate committed:
Merge branch 'master' into dev1_4_4

trufflepirate committed:
Edits to the UG

trufflepirate committed:
Update config.json

trufflepirate committed:
Updated Comments

trufflepirate committed:
Update ParserUtil.java

trufflepirate committed:
Travis

trufflepirate committed:
Travis tests fail but Gradle Tests pass

trufflepirate committed:
Fixing gradle tests

trufflepirate committed:
Fixing gradle tests

trufflepirate committed:
Fixing gradle tests

trufflepirate committed:
Fixing gradle tests

trufflepirate committed:
Fixing gradle tests

trufflepirate committed:
Fixing gradle tests

Ftywan committed:
added the testing instructions for manual testing

Ftywan committed:
Merge remote-tracking branch 'JJ/v1.4ugdg' into v1.4ugdg

trufflepirate committed:
Fixing gradle tests

trufflepirate committed:
Fixing gradle tests

trufflepirate committed:
Merge branch 'v1.4ugdg' of C:\Users\SEANY\Desktop\Year2 Sem1\CS2113T\main with conflicts.

trufflepirate committed:
Merge remote-tracking branch 'jjRepo/v1.4ugdg' into v1.4ugdg

trufflepirate committed:
Fixing gradle tests and Disabled old AB commands

trufflepirate committed:
Added manageJobs stuff

trufflepirate committed:
preferences

Commit:
Merge pull request #161 from trufflepirate/dev1_4_4

BugFixes and Tests

Commit:
Merge branch 'master' into v1.4ugdg

trufflepirate committed:
Some PPP work

trufflepirate committed:
Merge remote-tracking branch 'jjRepo/v1.4ugdg' into v1.4ugdg

trufflepirate committed:
More Updates to the DG and PPP

teojunjie committed:
Updated teojunjie ppp

trufflepirate committed:
Updated UG DG AND PPP

trufflepirate committed:
Updates to test cases

trufflepirate committed:
Changes to PPP

trufflepirate committed:
More PPP stuff

trufflepirate committed:
Changed UI PNG

trufflepirate committed:
PPP Changes

trufflepirate committed:
Fixing Travis

trufflepirate committed:
Shorten

teojunjie committed:
Merge branch 'v1.4ugdg' of https://github.com/teojunjie/main into v1.4ugdg

trufflepirate committed:
updated UG

teojunjie committed:
Merge branch 'v1.4ugdg' of https://github.com/teojunjie/main into v1.4ugdg

Psyf committed:
PPP check

Psyf committed:
Change some things in DG

Psyf committed:
Merge remote-tracking branch 'jjRemote/v1.4ugdg' into v1.4ugdg

Ftywan committed:
update the ppp

Psyf committed:
Big changes all around

Psyf committed:
cosmetic changes to AboutUs

Ftywan committed:
update the PPP

Ftywan committed:
Merge remote-tracking branch 'JJ/v1.4ugdg' into v1.4ugdg

trufflepirate committed:
Added login display

trufflepirate committed:
Some PPP work

teojunjie committed:
Merge branch 'v1.4ugdg' of https://github.com/teojunjie/main into v1.4ugdg

teojunjie committed:
Added in intro paragraph

teojunjie committed:
Cut down intro and other stuff to reduce to 10 pages

Commit:
Update psyf.adoc

Remove a redundant repitition

teojunjie committed:
Removed one small part from fileStorageSystem in dg

teojunjie committed:
Merge branch 'v1.4ugdg' of https://github.com/teojunjie/main into v1.4ugdg

trufflepirate committed:
Fixing UG bug discovered + manageJob error message

trufflepirate committed:
travis

Ftywan committed:
ppp work

trufflepirate committed:
Merge pull request #153 from teojunjie/v1.4ugdg

V1.4 Tenative Final Relase
+if there are no last minute bugs discovered

Commit:
Fix broken image

Commit:
Clarifications for sample load :: UG

Commit:
Sample data load clarifications :: DG

Commit:
PNG problems fixed

Seann -_-

Commit:
Travis fix

trufflepirate committed:
Fixed UG picture disaply errors

trufflepirate committed:
fixed view on github link

Commit:
Cosmetic UG change
